pkg/util/oom/oom_linux.go

```diff
@@ -75,6 +75,7 @@ func applyOOMScoreAdj(pid int, oomScoreAdj int) error {
 			err = fmt.Errorf("failed to apply oom-score-adj to pid %d (%v)", pid, err)
 			continue
 		}
+		defer f.Close()
```

Comment:
If the point of the loop is to retry, but then to error out if no loop
succeeded, we want to do:

```
if closeErr := f.Close(); closeErr != nil {
  err = fmt.Errorf(..., closeErr)
  continue
}
```

f.Close() is as valid an error condition as f.Write erroring

On Thu, Jul 21, 2016 at 3:41 AM, Peter Zhao <notifications@github.com>
wrote:

> In pkg/util/oom/oom_linux.go
> <https://github.com/kubernetes/kubernetes/pull/29266#discussion_r71659821>
> :
>
> > @@ -75,6 +75,7 @@ func applyOOMScoreAdj(pid int, oomScoreAdj int) error {
> >  			err = fmt.Errorf("failed to apply oom-score-adj to pid %d (%v)", pid, err)
> >  			continue
> >  		}
> > +		defer f.Close()
>
> Let me see. Something to be discussed:
> 1. The for loop tries twice to write file. If we return error directly
> when the first time writing file fails, there would be no chance to try the
> next time.
> 2. According to 1, we should continue instead of return. If f.Write
> returns error both the twice trying, we only need to care the last error,
> and it will be returned by the last sentence return err. If f.Write
> returns error the first time and nil second, we should ignore the first
> error also.
> 3. It seems that we may not care about the result of f.Close() ?
> /cc @smarterclayton <https://github.com/smarterclayton> . If there is
> something I misunderstand, feel free to point it out. Thanks!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/29266/files/8071b06b18a776eef8060b2c263752b3d56b47f5#r71659821>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p9_EYarIWxX_XkxWVmkv_RK8u_ebks5qXyKwgaJpZM4JQkT9>
> .
>